### PR TITLE
ISPN-5096 Avoid NPE beans when scanning for duplicates

### PR DESCRIPTION
**Pull request by vjuranek**
https://issues.jboss.org/browse/ISPN-5096

### CODE REVIEW

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity Infinispan :: Pull requests monitor [Build 8217/vjuranek:ISPN-5096](http://ci.infinispan.org/viewLog.html?buildId=22589&buildTypeId=bt9&guest=1) for 3755b11a54e4c417cd9b0a18a7ffd0221afa00f9
Tests: 11216, 2 failed (2 new), 23 ignored. Build time: 01:51:13

Failed tests detected
* (new) TestSuite: org.infinispan.partitionhandling.EntryRetrieverReplPartitionHandlingTest.testRetrievalWhenPartitionIsDegrated


* (new) org.infinispan.server.test.query.ManualIndexingIT.testManualIndexing

**Comment by galderz** [comment]
@vjuranek Thoughts on @danberindei's comments?

File: cdi/src/main/java/org/infinispan/cdi/util/defaultbean/DefaultBeanExtension.java [hunk 1]
```diff
@@ -340,11 +340,12 @@ void afterBeanDiscovery(@Observes AfterBeanDiscovery event, BeanManager manager)
             if (bean == null) {
                 bean = defaultProducerFields.get(e.getKey());
             }
-            if (duplicateDetectionMap.containsKey(e.getValue())) {
+            if (bean != null && duplicateDetectionMap.containsKey(e.getValue())) {
```
Comment (slaskawi) [inline cdi/src/main/java/org/infinispan/cdi/util/defaultbean/DefaultBeanExtension.java]: Please correct me if I'm wrong, but it happened when testing 2 uberjars together (which in turn cased this issue: https://issues.jboss.org/browse/ISPN-5092)?

If I'm correct - we won't be able to support such setup. Here is a relevant part from CDI spec [1]: `If a bean class is deployed in two different bean archives, non-portable behavior results. Portable applications must deploy each bean class in no more than one bean archive.`

[1] http://docs.jboss.org/cdi/spec/1.0/html_single/#beanarchive

Comment (vjuranek) [inline cdi/src/main/java/org/infinispan/cdi/util/defaultbean/DefaultBeanExtension.java]: @slaskawi yes, it happened when testing 2 uberjars together. I'm not 100% sure if this was the root cause (haven't had time to find a reproducer yet), but IMHO it's very likely to be so. I'm going to close this PR and I'll reopen when spot it again in supported use case (or found out, that the root cause was different)

**Comment by vjuranek** [comment]
Fixes a case which is not supported (two uberjars deployed together), closing.

**Pull request closed without merging**